### PR TITLE
Validate OCR presence

### PR DESCRIPTION
**Pull request by rkondratowicz**
TODO: pass doc with OCR to `OcrValidator` (currently it finds the document to validate in envelope)

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/FailedDocUploadProcessor.java [hunk 1]
```diff
@@ -56,6 +57,7 @@ public FailedDocUploadProcessor(
         EnvelopeRepository envelopeRepository,
         ProcessEventRepository eventRepository,
         ContainerMappings containerMappings, // NOSONAR
+        OcrPresenceValidator ocrPresenceValidator, // NOSONAR
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/FailedDocUploadProcessor.java]: can't wait to get rid of that inheritance...

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/OcrPresenceValidator.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    public static final String MISPLACED_OCR_MSG = "OCR on document of invalid type";
+
+    /**
+     * Checks whether OCR data is on the correct document.
+     *
+     * @return Document with OCR.
+     */
+    public Optional<InputScannableItem> assertHasProperlySetOcr(List<InputScannableItem> docs) {
+
+        throwIf(docs.stream().filter(it -> it.ocrData != null).count() > 1, MULTIPLE_OCR_MSG);
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/OcrPresenceValidator.java]: we will need more information in the message like document control number which had the OCR and also if we can document type/subtype that would be bonus.

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/OcrPresenceValidator.java]: how about I catch, log where this is being thrown, and rethrow?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/OcrPresenceException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.exceptions;
+
+public class OcrPresenceException extends InvalidEnvelopeException {
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/OcrPresenceException.java]: I'd squeeze `Validation` work into this name

**Pull request merged**